### PR TITLE
Store pending unspent notes

### PR DESCRIPTION
**Pull request by panleone**
Store pending unspent notes (for example changes)

### CODE REVIEW

**Comment by panleone** [comment]
ready for review

**Comment by Duddino** [comment]
Change the name of the commit please

File: js/pivx_shielding.js [hunk 1]
```diff
@@ -258,6 +274,7 @@ export class PIVXShielding {
    */
   discardTransaction(txid) {
     this.pendingSpentNotes.clear(txid);
+    this.pendingUnspentNotes.clear(txid);
```
Comment (Duddino) [inline js/pivx_shielding.js:277]: This means the unspent notes will be cleared when calling finalizeTransaction too

File: js/pivx_shielding.js [hunk 2]
```diff
@@ -199,6 +215,17 @@ export class PIVXShielding {
     return this.unspentNotes.reduce((acc, [note]) => acc + note.value, 0);
   }
 
+  /**
+   * Return number of pending satoshis of the account
+   */
+  getPendingBalance() {
+    let total = 0;
```
Comment (Duddino) [inline js/pivx_shielding.js:222]:
```suggestion
    Array.from(this.pendingUnspentNotes.values()).flat().reduce((acc, v) => acc + v, 0)
```

**Pull request merged**